cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/app/etl/batch/ETLMapReduce.java

```diff
@@ -299,6 +383,23 @@ public void destroy() {
     }
   }
 
+  private static class ErrorSink<IN, KEY_OUT, VAL_OUT> {
```

Comment:
map method has ``MapreduceContext`` while we want ``MapreduceTaskContext`` in order to be able to write to the particular error dataset. that's why we need this abstraction.